### DISCUSSION TITLE
Assigning the same facility id to multiple racks

### DISCUSSION THREAD

**Discussion by steffzio** [netbox-community/netbox#15035] [Q&A]
I have one site with multiple locations and within those locations I often have multiple racks. When I assign the same facility IDs to those racks within the same location I get this error. 
![image](https://github.com/netbox-community/netbox/assets/11305259/80b083a5-c5bb-4fb1-8af8-939cffebf5c5)
I would expect this to work since the racks all have unique names, and it's not uncommon to have multiple racks in the same location.

Could anybody confirm if this is a bug or expected behavior?

**Comment by candlerb** [comment]
I believe that "Facility ID" is meant to mean "the unique name or ID that this rack is known as by the provider". You don't have to set it at all, but if you do, it has to be unique.

If by "Facility" you are thinking "Data center ID", then this would be the Facility attribute of the Site object.

**Reply by steffzio** [reply]
Well I guess that would make sense from a large colocation data center perspective. 

I'm using one site for our entire campus and locations for different buildings and/or rooms. My plan was to use Facility ID as room number since we cannot assign those to a location where it would make to most sense. I will solve it be either moving all location to sites or by using custom fields. 

The documentation should be updated to mention that the facility id has to be unique as it does with asset tags for example.

**Reply by candlerb** [reply]
> I'm using one site for our entire campus and locations for different buildings and/or rooms. My plan was to use Facility ID as room number since we cannot assign those to a location where it would make to most sense

Locations have names (which are unique with the scope of their parent): the room number would make a perfectly good location name IMO. They also have a description, and if you want anything else you can add a custom field.

Therefore, I think your original plan of using one site for the whole campus, and locations for buildings and rooms within buildings, is fine.

> The documentation should be updated to mention that the facility id has to be unique as it does with asset tags for example.

The [documentation](https://docs.netbox.dev/en/stable/models/dcim/rack/#facility-id) currently says:

*"An alternative identifier assigned to the rack e.g. by the facility operator. This is helpful for tracking datacenter rack designations in a colocation facility."*

I guess "unique" could be added there, but it does make it clear that it's an identifier for the rack itself, and it would surely be confusing if two different racks were allowed to have the same identifier.

In the rack edit page, it says "Locally-assigned identifier". Maybe it could be clarified that it is a locally-assigned identifier for the rack itself; but since you're in a page where you're editing details of the rack, that's how I already interpreted it.
